### ISSUE TITLE
Default Research to 1

### ISSUE THREAD

**Issue by JavidPack** [bluemagic123/tModLoader#3228]
### Do you intend to personally contribute/program this feature?

Yes

### I would like to see this change made to improve my experience with

Mod capability as a Modder

### Description

Journey mode has a feature where players can "research" an item. Once researched, a player can get a copy of the item whenever they want.

Aside from [unobtainable items](https://terraria.wiki.gg/wiki/Unobtainable_items), all vanilla items are researchable. Most items require 1 of the item to "research" it. https://terraria.wiki.gg/wiki/Journey_Mode#Research

Currently, modders need to specify this in SetStaticDefaults:
```
public override void SetStaticDefaults() {
	Item.ResearchUnlockCount = 1;
}
```

If a modder doesn't, then the item can't be researched. This is an issue, since many modders forget to add this line of code, and then users on journey mode have a suboptimal experience. Since most items need 1 item, most items will have these lines of code.

### What does this proposal attempt to solve or improve?

The proposal is to default all modded items to 1 instead of 0. All items will be researchable and only modders interested in balancing for journey mode would need to ever write the `Item.ResearchUnlockCount =` code. This will also allow many examplemod examples to be less cluttered. 

The only issue is with items that should disappear on pickup, modders with those type of items will need to add `Item.ResearchUnlockCount = 0;` to those items.

**Comment by JavidPack** [comment]
Expose ContentSamples.AddItemResearchOverride maybe?

**Comment by JavidPack** [comment]
Also: 0 is not the same as not existing internally, 0 is reportedly a bug. Code modders call will need to remove the key from the dictionary.

**Comment by Setnour6** [comment]
I do have to argue a bit with this feature, because with everything set to 1 by default, it makes it harder to keep track of what items need to be changed internally for implementing the research functions. For example, let's say that everything from example mod is used to make your own mod, and there are only a handful of non-weapon items that have `ResearchUnlockCount` set to 3. In a month from now, you might forget what items need to be changed or updated internally to provide a constant value of research.
The unprovided research count, although prevents the items from being researched, encourages modders to provide a research value for their items.
You could try to add a message to each mod item in-game that says "research values for this mod item have not been provided (this is a message to the developer)", or have a warning box appear on the mod itself in the mod list, saying "research values have not been provided for some items (this is a message to the developer)", and players can click on the warning box to show a list of items, with external and/or internal names appearing, so modders can work on providing research values for the specified items.

**Also**, because I want to keep this in one message:

In `Item.TML.Utils`, would this work?
```cs
public int ResearchUnlockCount {
		get => CreativeItemSacrificesCatalog.Instance.SacrificeCountNeededByItemId.TryGetValue(type, out int result) ? result : 1;
		set => CreativeItemSacrificesCatalog.Instance.SacrificeCountNeededByItemId[type] = value;
	}
```
_Note the `1` at the end of the accessor_

Or are there other things that will need to change in order to fully reflect the intended proposal?

**Comment by Chicken-Bones** [comment]
If I understand correctly, your workflow was going through all your `ModItem` files to see if they're missing a `ResearchUnlockCount = ...` line and deciding which of these were _meant_ to be unresearchable and which were just forgotten about.

If you just copied `ResearchUnlockCount = 1` from another item or example, then you wouldn't be able to tell whether the research value was intentional or not.

I'm not sure anyone would want to review the vast majority of `ResearchUnlockCount = 1` items in their mod periodically, so I don't think having a "this default set by TML" or such would be helpful, since for the majority, the default being set by TML is the intentional final state.

I it may be better for you to solve this via a process in your team, such as leaving a `// TODO determine appropriate research count` whenever you make a `ModItem`

**Comment by JavidPack** [comment]
This was resolved in #3242
